pkg/cli/user.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"context"
+	"os"
+
+	"github.com/vmware/govmomi/ssoadmin"
+	"github.com/vmware/govmomi/ssoadmin/types"
+	"github.com/vmware/govmomi/sts"
+	"github.com/vmware/govmomi/vim25/soap"
+)
+
+type User struct {
+	id string
+	types.AdminPersonDetails
+	password string
+	solution types.AdminSolutionDetails
+	role     string
+}
+
+// CreateUserFunc is function to create person user or solution user
+type CreateUserFunc func(c *ssoadmin.Client) error
+
+func (u *User) Run(ctx context.Context, c *ClientOption, fn CreateUserFunc) error {
+
+	vc, err := c.GetClient()
+	if err != nil {
+		return err
+	}
+
+	ssoClient, err := ssoadmin.NewClient(ctx, vc)
+	if err != nil {
+		return err
+	}
+
+	// SSO admin server has its own session manager. There is no SSO token persistence yet, so just use an env
```

Comment:
The persistence comment only applied to govc - sessions are persisted by default by govc in `~/.govmomi/sessions`